src/pages/index.js

```diff
@@ -34,50 +36,126 @@ const enableValidation = (formSettings) => {
-  cardGallery.addItem(data);
-  cardForm.close();
+  cardForm.submitButton.textContent = "Saving...";
+  api
+    .addNewCard({ name: data.name, link: data.link })
+    .then((res) => (res.ok ? res.json() : res.status))
+    .then((res) => {
+      cardGallery.addItem(res);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      cardForm.submitButton.textContent = "Create";
+      cardForm.close();
+    });
 });
 cardForm.setEventListeners();
 
 const profileForm = new PopupWithForm("form[name='profile-form']", (data) => {
-  userInfo.setUserInfo(data);
-  profileForm.close();
+  profileForm.submitButton.textContent = "Saving...";
+  api
+    .editUserInfo({ name: data.name, job: data.job })
+    .then((res) => (res.ok ? true : res.status))
+    .then(() => {
+      userInfo.setUserInfo({ name: data.name, about: data.job });
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      profileForm.submitButton.textContent = "Save";
+      profileForm.close();
+    });
 });
 profileForm.setEventListeners();
 
+const confirmDeletePopup = new PopupDeleteConfirmation(
+  "form[name='delete-form']"
+);
+confirmDeletePopup.setEventListeners();
+
 const picturePopup = new PopupWithImage(".popup__picture");
 picturePopup.setEventListeners();
 
 /* *********************************** */
 /* Declaration of card gallery section */
 /* *********************************** */
-const cardGallery = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      const card = new Card(data, ".card-template", (data) => {
-        picturePopup.open(data);
+const cardGallery = new Section((data) => {
+  const card = new Card(
+    data,
+    ".card-template",
+    (data) => {
+      picturePopup.open(data);
+    },
+    () => {
+      confirmDeletePopup.setSubmitAction(() => {
+        api.deleteCard(data._id);
+        card.deleteCard();
+        confirmDeletePopup.close();
       });
-      const cardElement = card.generateCard();
-      cardGallery.container.prepend(cardElement);
+      confirmDeletePopup.open();
     },
-  },
-  ".gallery__cards"
-);
-
-cardGallery.renderItems();
+    () => {
+      const cardLikeIcon = card.card.querySelector(".card__like");
+      if (card.liked) {
+        api
+          .removeLike(card.id)
+          .then((res) => (res.ok ? true : res.status))
+          .then(() => {
+            cardLikeIcon.classList.remove("card__like_active");
+            card.liked = false;
+          })
+          .catch((err) => console.error(err));
+      } else {
+        api
+          .addLike(card.id)
+          .then((res) => (res.ok ? true : res.status))
+          .then(() => {
+            cardLikeIcon.classList.add("card__like_active");
+            card.liked = true;
+          })
+          .catch((err) => console.error(err));
+      }
+    }
+  );
+  const cardElement = card.generateCard();
+  cardGallery.container.append(cardElement);
```

Comment:
`  cardGallery.container.append(cardElement);
`

* Please, create `prependItem` and `appendItem` inside `Section`. They should add cards into the container **inside** the class rather than outside it